packages/dd-trace/src/dogstatsd.js

```diff
@@ -48,16 +56,38 @@ class Client {
     }
   }
 
-  _send (address, family, buffer) {
+  _sendHttp (queue, address, family, options) {
+    const buffer = Buffer.concat(queue)
+    request(buffer, options, (err) => {
+      if (err) {
+        log.error('HTTP error from agent: ' + err.stack)
+        if (err.status) {
+          // Inside this if-block, we have connectivity to the agent, but
+          // we're not getting a 200 from the proxy endpoint. Fall back to
+          // UDP and try again.
+          this._httpOptions = null
+          this._sendAll(queue, address, family)
+        }
+      }
+    })
+  }
+
+  _sendUdp (address, family, buffer) {
     const socket = family === 6 ? this._udp6 : this._udp4
 
     log.debug(`Sending to DogStatsD: ${buffer}`)
 
     socket.send(buffer, 0, buffer.length, this._port, address)
   }
 
-  _sendAll (queue, address, family) {
-    queue.forEach((buffer) => this._send(address, family, buffer))
+  _sendAll (queue, address, family, options) {
+    if (options) {
```

Comment:
This is unnecessarily confusing, this condition would be better suited in the existing condition in flush so that it only happens once instead of multiple times. There are no scenarios that would result in an HTTP call except when there are HTTP options.

Reply:
I've cleaned this up quite a bit in the next push.